applications/app/controllers/ReaderRevenueController.scala

```diff
@@ -0,0 +1,32 @@
+package controllers
+
+import common.{ImplicitControllerExecutionContext, Logging}
+import model._
+import play.api.mvc.{Action, AnyContent, BaseController, ControllerComponents}
+import services.S3
+
+import scala.concurrent.duration._
+import conf.Configuration.readerRevenue._
+import model.Cached.{RevalidatableResult}
+
+
+class ReaderRevenueController(val controllerComponents: ControllerComponents)(implicit context: ApplicationContext)
+  extends BaseController with ImplicitControllerExecutionContext with Logging {
+
+  private def getContributionsBannerDeployLog(): Option[String] = {
+    S3.get(contributionsBannerDeployLogKey)
+  }
+
+  private def bannerDeployLogUnavailable() = {
+    log.warn(s"Could not get reader revenue contributions-banner deploy log from s3")
+    NoCache(NotFound)
```

Comment:
Maybe you could still cache the 404, perhaps for a shorter time - so that if there is some kind of S3 outage we don't start getting loads of origin traffic?